### PR TITLE
Add more tests and related fixes

### PR DESCRIPTION
**Pull request by achow101**
Adds tests for nearly all commands for all devices and related fixes. Only `restore` for `trezor` and `keepkey` are untested as I could not get them to work. All other commands for all devices have tests and related bug fixes.

### CODE REVIEW

File: hwilib/base58.py [hunk 1]
```diff
@@ -87,23 +87,6 @@ def to_address(b, version):
...
-        return to_address(pkh, b'\x6f')
-    else:
-        return to_address(pkh, b'\x00')
-
-def xpub_to_address(xpub, testnet=False):
-    data = decode(xpub)
-    pubkey = data[-37:-4]
-    return pubkey_to_address(pubkey, testnet)
-
-def xpub_to_pub_hex(xpub):
```
Comment (Sjors) [inline hwilib/base58.py]: I still need `xpub_to_pub_hex` in #117, because the inferred descriptors from Core use the pubkey in hex.

**Comment by Sjors** [comment]
I'm all for more tests. Light review of d097646 only found one problem, but I'm not very familiar with the test suite.

File: hwilib/base58.py (hunk 1 above)
Comment (achow101) [reply hwilib/base58.py]: I've added it back in

**Comment by achow101** [comment]
Merging as tests pass

**Pull request merged**